### PR TITLE
2639 item merging requisition line

### LINKED ISSUE #2639: Item merging: requisition_line
## Is your feature request related to a problem? Please describe 👀

To fulfill item merging we need to implement usage of `item_link` for invoice_lines.

## Describe the solution you'd like 🎁
Base: [feature/merging](https://github.com/msupply-foundation/open-msupply/tree/feature/merging)

Using changes in the [stock_line PR](https://github.com/msupply-foundation/open-msupply/pull/2547/files#diff-63618ddd9539c21a7c93d1c9432eb912541ea89aa94f0b73c7052550ba01770f) as a guide (maybe checkout the latest version, not that specific commit):
- [ ] Add `requisition_line.item_link_id` in migrations
- [ ] Drop `requisition_line.item_id`
- [ ] Update any sql views that relied on `requisition_line.item_id`.
- [ ] Update requisition_lines at the repository layer to return items based on `requisition_line.item_link_id` rather than `requisition_line.item_id`

### Additional context 💌

Whoops somehow missed this one 😉.

### Moneyworks Jobcode 🧰

OMS:MERGE

### PR DESCRIPTION
**Pull request by Chris-Petty**
Fixes #2639

# 👩🏻‍💻 What does this PR do? 

- Adds item merge changes to the requisition_line repos 

# 🧪 How has/should this change been tested? 

Tests pass

### CODE REVIEW

**Review by andreievg — COMMENTED** [review]
Looks good, keen to approve, but should it have a test somewhere here: https://github.com/msupply-foundation/open-msupply/blob/eba8f6ec209fc72b6f86b060b1c1ff57fcd53b83/server/repository/src/migrations/v1_06_00/mod.rs#L239

**Comment by Chris-Petty** [comment]
@andreievg I have added z testz

**Comment by Chris-Petty** [comment]
@andreievg bump

File: server/repository/src/migrations/v1_06_00/mod.rs [hunk 1]
```diff
@@ -117,19 +118,52 @@ async fn migration_1_06_00() {
...
+        requisition_line (id) {
+            id -> Text,
+            requisition_id -> Text,
+            item_link_id -> Text,
+            requested_quantity -> Integer,
+            suggested_quantity -> Integer,
+            supply_quantity -> Integer,
+            available_stock_on_hand -> Integer ,
+            average_monthly_consumption -> Integer,
+            snapshot_datetime -> Nullable<Timestamp>,
```
Comment (andreievg) [inline server/repository/src/migrations/v1_06_00/mod.rs:139]: For the future, I think we can leave Nullable out and actually, the requisition_line we are testing, only really needs to have id and item_link_id (when unless we are also using the data time to do upserting)

**Review by andreievg — APPROVED** [review]
Thanks @Chris-Petty looks great

**Pull request merged**